test/docfx.Test/lsp/LanguageServerTestClient.cs

```diff
@@ -169,14 +193,26 @@ private async Task<ILanguageClient> InitializeClient(string workingDirectory, Pa
                         _notificationSync.TrySetException(new InvalidOperationException(message.Message));
                     }
                 })
+                .OnRequest("docfx/getCredential", async (GetCredentialParams @params) =>
+                {
+                    if (_credentialRefreshHandler == null)
```

Comment:
The client may receive `docfx/getCredential` request and hit this line before it had the change to process `expectGetCredentialRequest` and set the delegate there. This could use the same synchronization mechanism as notification, basically this `OnRequest` callback puts a `(JToken request, TaskCompletionSource<JToken> response)` to a member variable, and in the `ProcessCommand` method, first call `await SynchronizeNotifications();`, then return the result through `TaskCompletionSource<JToken>`.